packages/typescript-estree/tools/test-utils.ts

```diff
@@ -27,24 +11,24 @@ export function parseCodeAndGenerateServices(
 /**
  * Returns a function which can be used as the callback of a Jest test() block,
  * and which performs an assertion on the snapshot for the given code and config.
- * @param {string} code The source code to parse
- * @param {TSESTreeOptions} config the parser configuration
- * @param {boolean} generateServices Flag determining whether to generate ast maps and program or not
- * @returns {jest.ProvidesCallback} callback for Jest it() block
+ * @param code The source code to parse
+ * @param config the parser configuration
+ * @param generateServices Flag determining whether to generate ast maps and program or not
+ * @returns callback for Jest it() block
  */
 export function createSnapshotTestBlock(
   code: string,
   config: TSESTreeOptions,
   generateServices?: true,
-): () => void {
+): jest.ProvidesCallback {
   /**
-   * @returns {Object} the AST object
+   * @returns the AST object
    */
   function parse(): parser.TSESTree.Program {
     const ast = generateServices
       ? parser.parseAndGenerateServices(code, config).ast
       : parser.parse(code, config);
-    return getRaw(ast);
+    return omitDeep(ast);
```

Comment:
I would prefer if this wasn't used like this. It looks pretty confusing.

It'd be better if `omitDeep` was aliased and strictly typed with one arg, so that nobody accidentally uses the other args, mutating the object instead of straight copying it.

```ts
function deeplyCopy<T>(thing: T): T {
  return omitDeep(thing);
}
```